client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -26,35 +26,82 @@ import './style.scss';
 
 export class SiteVerticalsSuggestionSearch extends Component {
 	static propTypes = {
+		autoFocus: PropTypes.bool,
 		charsToTriggerSearch: PropTypes.number,
-		initialValue: PropTypes.string,
+		defaultVertical: PropTypes.object,
 		onChange: PropTypes.func,
 		placeholder: PropTypes.string,
-		shouldShowPopularTopics: PropTypes.func,
 		searchValue: PropTypes.string,
+		shouldShowPopularTopics: PropTypes.func,
 		verticals: PropTypes.array,
-		defaultVertical: PropTypes.object,
 	};
 
 	static defaultProps = {
+		autoFocus: false,
 		charsToTriggerSearch: 1,
-		initialValue: '',
+		defaultVertical: {},
 		onChange: noop,
 		placeholder: '',
-		shouldShowPopularTopics: noop,
 		searchValue: '',
+		shouldShowPopularTopics: noop,
 		verticals: [],
-		defaultVertical: {},
 	};
 
 	constructor( props ) {
 		super( props );
 		this.state = {
 			railcar: this.getNewRailcar(),
 			candidateVerticals: [],
+			isSuggestionSelected: false,
+			searchValue: props.searchValue,
```

Comment:
I'm going to touch one of the hardest CS problem here: Could we name these two state / props differently so we can distinguish them better?

I personally found a few times that I misunderstood the code because I misread `this.state.searchValue` as `this.props.searchValue`, and vice versa. Thus, it would be great if we can have better naming here.

I feel `searchValue` for the state is quite nice, but what would match the meaning of the current `this.props.searchValue` better? In my previous run of refactoring, it becomes a props because the component doesn't hold the input box value itself anymore. Now that the react state one is back with a legit reason (I think it's because debouncing. Please feel free to correct me if I'm wrong),  it's time to revisit the naming and the necessity of this prop.

I don't really know the best direction of this, so this is more of a brain dump comment than a request to fix. Please feel free to keep it this way and we can revisit it in the future.

Reply:
I think it was a product of the rebase after the state tree updates.

I'll change it back to my original name `initialValue`, which describes the prop more accurately now since we're using the internal state to track updates.

Cheers!